temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -159,6 +159,30 @@ public ActivityOptions getDefaultActivityOptions() {
     return activityOptionsMap;
   }
 
+  public void setDefaultActivityOptions(ActivityOptions defaultActivityOptions) {
+    this.defaultActivityOptions =
+        (this.defaultActivityOptions == null)
+            ? defaultActivityOptions
+            : this.defaultActivityOptions
+                .toBuilder()
+                .mergeActivityOptions(defaultActivityOptions)
+                .build();
+  }
+
+  public void setActivityOptions(Map<String, ActivityOptions> activityMethodOptions) {
+    if (this.activityOptionsMap == null) {
+      this.activityOptionsMap = activityMethodOptions;
```

Comment:
Let's make a copy of the map to ensure that it is immutable. I think we should do it in the implementation options as well.